environment/environment_test.go

```diff
@@ -0,0 +1,66 @@
+package environment
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/nutanix-cloud-native/prism-go-client/environment/types"
+	. "github.com/onsi/gomega"
+)
+
+type testProvider struct {
+	// single management endpoint
+	mgmtEndpoint *types.ManagementEndpoint
+	// variables as JSON doc
+	vars map[string]interface{}
+}
+
+func (prov *testProvider) GetManagementEndpoint(
+	topology types.Topology,
+) (*types.ManagementEndpoint, error) {
+	return prov.mgmtEndpoint, nil
+}
+
+func (prov *testProvider) Get(topology types.Topology, key string) (
+	interface{}, error,
+) {
+	if val, ok := prov.vars[key]; ok {
+		return val, nil
+	}
+	return nil, types.ErrNotFound
+}
+
+func TestEnvironment(t *testing.T) {
+	g := NewWithT(t)
```

Comment:
nit: consider using testify. If using ginkgo, please don't `.` import for readability :)

Reply:
I might do more than assertions :) I like the fact that Gomega can do async asserts. Also in k8s ecosystem everyone seemed to have standardized on Ginkgo + Gomega, including Kubernetes itself.

`.` import is considered idiomatic in Ginkgo and Gomega, but have now strong opinions about it.

Reply:
I did not make this change as code would be very verbose. Gomega (and Ginkgo for that matter) can be understood as DSLs. Functions are used like natural language. With `gomega.` prefix readability is really impacted.

Reply:
While I don't agree with the unfortunate ginkgo-everything landscape of Golang development today, I won't be gating the PR on this.

Reply:
I'm falling in line with what k8s is doing. We use Ginkgo/Gomega in CNDS and k8s-e2e-test as well. I understand it's overkill for simple unit tests but simplifies writing more complex integration tests.